src/checkout-buttons/strategies/masterpass-button-strategy.ts

```diff
@@ -0,0 +1,144 @@
+            ]
+        ).then(([statePayment, stateCheckout]) => {
+                this._paymentMethod = statePayment.paymentMethods.getPaymentMethod(methodId);
+                if (!this._paymentMethod || !this._paymentMethod.initializationData.checkoutId) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                this._checkout = stateCheckout.checkout.getCheckout();
+                if (!this._checkout || !this._checkout.cart.id) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                return this._masterpassScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(masterpass => {
+                        this._masterpassClient = masterpass;
+                        this._createSignInButton(masterpassOptions);
+                    });
+        })
+        .then(() => super.initialize(options));
+    }
+
+    deinitialize(options: CheckoutButtonOptions): Promise<void> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        this._paymentMethod = undefined;
+        this._checkout = undefined;
+
+        if (this._signInButton && this._signInButton.parentNode) {
+            this._signInButton.removeEventListener('click', this._handleWalletButtonClick);
+            this._signInButton.parentNode.removeChild(this._signInButton);
+            this._signInButton = undefined;
+        }
+
+        return super.deinitialize(options);
+    }
+
+    private _createSignInButton(masterpassOptions: MasterpassButtonInitializeOptions): void {
+        const { container } = masterpassOptions;
+        const buttoncontainer = document.querySelector(`#${container}`);
+
+        if (!buttoncontainer) {
+            throw new Error('Need a container to place the button');
+        }
+
+        const button = document.createElement('input');
+
+        button.type = 'image';
+        button.src = 'https://static.masterpass.com/dyn/img/btn/global/mp_chk_btn_160x037px.svg';
+        buttoncontainer.appendChild(button);
+        this._signInButton = button;
+        this._signInButton.addEventListener('click', this._handleWalletButtonClick);
+    }
+
+    private _createMasterpassPayload(): MasterpassCheckoutOptions {
+        return {
+            checkoutId: this.paymentMethod.initializationData.checkoutId,
+            allowedCardTypes: this.paymentMethod.initializationData.allowedCardTypes,
+            amount: this.checkout.cart.cartAmount.toString(),
+            currency: this.checkout.cart.currency.code,
+            cartId: this.checkout.cart.id,
+            suppressShippingAddress: true,
+        };
+    }
+
+    @bind
+    private _handleWalletButtonClick(): void  {
+        const payload = this._createMasterpassPayload();
+        this.masterpassClient.checkout(payload);
+    }
+
+    private get masterpassClient(): Masterpass {
```

Comment:
Firstly, we actually don't allow computed properties in this project. `get/set` are forbidden because it is very easy to put logic in there that's computationally expensive or induces side effects (for more information, please see this [article](https://github.com/airbnb/javascript#accessors--no-getters-setters)).

Secondly, all private properties and methods need to be prefixed with `_`. This is because, although we're using TypeScript, the source ultimately gets compiled into JS and consumed by JS clients. So we need a way to indicate to the consumers that these properties and methods are not supposed to be used.

If you really want to go with this pattern, you need to name this method as `_getMasterClient`. But often we just inline the guard statement unless there's a real need (i.e.: you need to access the property multiple times).